### ISSUE TITLE
False positive `reportUnnecessaryContains` error when `self` is used

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6329]
The following code generates a false positive:

```python
# pyright: reportUnnecessaryContains=true

class Enum1(Enum):
    a = "a"
    b = "b"
    c = "c"

    @property
    def is_ab(self):
        return self in (Enum1.a, Enum1.b)

    @property
    def is_c(self):
        return self not in (Enum1.a, Enum1.b)
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.335, which I just published. It will also be included in a future release of pylance.
